LoopKit/DosingStrategy.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  DosingStrategy.swift
+//  LoopKit
+//
+//  Created by Pete Schwamb on 6/27/22.
+//  Copyright © 2022 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+
+public enum DosingStrategy: Int, CaseIterable, Codable {
+    case tempBasalOnly
+    case automaticBolus
+}
```

Comment:
`tempBasalOnly` seems to imply that there should be a dosing strategy that includes both a temp basal and bolus. If this isn't the case, why not remove `Only`. Also, is there a need to distinguish between automatic temp basal and a manually entered temp basal? Similarly, why isn't there a manually entered bolus strategy? If this is supposed to represent all the dosing strategies of Loop, it seems to be lacking.

Reply:
Automatic bolus strategy does include both temp basal and bolus. These names are based on the existing common usage names for the automatic dosing strategies in DIY.  Dosing strategy is a name for the *strategy* that Loop uses to attempt to bring BG into range, which can involve a variety of different dose *types*. The strategy has dose types in the name, just because that's how we've been referring to them for a long time, and doesn't need to, nor should include a summation of all the possible dose *types* that the strategy utilizes. The names primarily reflect the differences in the strategies.

Reply:
yeah... I see that it is cryptically limited to `automatic` dosing strategies, but general dosing strategies.

Reply:
I still think these names could be clearer. For example, does automaticBolus only include automatic bolus dosing and not temp basal? The current name seems to imply that.

Reply:
I was trying to explain that the possible values of this field are describing the "name" of the strategy, which happens to be named after the main dosing types they employ, but the values here are *not* intended to be way to control specifics about dosing. Each strategy differs in many aspects other than the primary dose type they use. 

If you had an Option Set of different dose types, and enabled 'bolus', but not 'temp basal', there is no "strategy" in loop that matches that.  Also, you could in theory have different dosing strategies that used the same set of dose types. In that case, the names would have to use something other than dose type to differentiate them. For now, the main (but not only) differences between the two dosing strategies that Loop offers is the dose type.